app/src/main/java/id/rizmaulana/covid19/ui/base/BaseViewItem.kt

```diff
@@ -1,5 +1,8 @@
 package id.rizmaulana.covid19.ui.base
 
+import id.rizmaulana.covid19.ui.adapter.ItemTypeFactory
+import id.rizmaulana.covid19.ui.adapter.ItemTypeFactoryImpl
+
 interface BaseViewItem {
-    fun layoutResId(): Int
+    fun typeOf(itemFactory: ItemTypeFactory): Int
```

Comment:
I think we should make it like this @fajarnuha 

`fun typeOf(itemFactory: ItemTypeFactory): Int = itemFactory.type(this)`

with this we don't even need to **override** it on it inheritor

Before:
```

data class OverviewItem(
    val confirmed: Int = 0,
    val recovered: Int = 0,
    val deaths: Int = 0
): BaseViewItem {
    override fun typeOf(itemFactory: ItemTypeFactory): Int = itemFactory.type(this)
}

```
After:

```
data class OverviewItem(
    val confirmed: Int = 0,
    val recovered: Int = 0,
    val deaths: Int = 0
): BaseViewItem
```

Reply:
oh i just knew kotlin interface can have a body, interesting